src/cascade/model/smooth_grid.py

```diff
@@ -0,0 +1,153 @@
+    """Slices to access priors with Distribution objects.
+    Each PriorView has one mulstd, corresponding with its kind.
+    """
+    def __init__(self, kind, ages, times):
+        self._kind = kind
+        super().__init__(
+            (ages, times),
+            columns=["density", "mean", "std", "lower", "upper", "eta", "nu", "name"])
+        # Let the base class make extra mulstds and delete them.
+        for del_kind in PriorKindEnum:
+            if del_kind.name != kind:
+                del self._mulstd[del_kind.name]
+
+    def age_time_diff(self):
+        """Iterate over (age, time) in the grid."""
+        yield from zip(
+            np.repeat(self.ages, len(self.times)),
+            np.tile(self.times, len(self.ages)),
+            np.repeat(np.ediff1d(self.ages, to_end=inf), len(self.times)),
+            np.tile(np.ediff1d(self.times, to_end=inf), len(self.ages)),
+        )
+
+    @property
+    def mulstd_prior(self):
+        """Standard deviation multiplier as a Prior."""
+        # The base class, AgeTimeGrid, has a dictionary of three mulstds.
+        # The prior grid uses only one of them.
+        return prior_distribution(self._mulstd[self._kind].iloc[0])
+
+    @mulstd_prior.setter
+    def mulstd_prior(self, value):
+        to_set = value.parameters()
+        to_assign = [to_set[setp] if setp in to_set else nan for setp in self.columns]
+        self._mulstd[self._kind].loc[:, self.columns] = to_assign
+
+    def __getitem__(self, at_slice):
+        return prior_distribution(super().__getitem__(at_slice).iloc[0])
+
+    def __setitem__(self, at_slice, value):
+        """
+        Args:
+            at_slice (slice, slice): What to change, as integer offset into ages and times.
+            value (priors.Prior): The prior to set, containing dictionary of
+                                  parameters.
+        """
+        to_set = value.parameters()
+        super().__setitem__(at_slice, [to_set[setp] if setp in to_set else nan for setp in self.columns])
+
+    def apply(self, transform):
+        for idx, row in self.grid.loc[:, self.columns + ["age", "time"]].iterrows():
+            new_distribution = transform(row.age, row.time, prior_distribution(row))
+            to_set = new_distribution.parameters()
+            self.grid.iloc[idx, self.columns] = [
+                to_set[setp] if setp in to_set else nan for setp in self.columns]
+
+
+class SmoothGrid:
+    def __init__(self, age_time_grid):
+        """
+        The Smooth Grid is a set of priors on an age-time grid.
+
+        Args:
+            age_time_grid (Tuple(set,set)): The supporting grid.
+        """
+        self.ages = np.sort(np.array(age_time_grid[0], dtype=np.float))
+        self.times = np.sort(np.array(age_time_grid[1], dtype=np.float))
+        self._view = dict()
+        for create_view in PriorKindEnum:
+            self._view[create_view.name] = _PriorView(create_view.name, self.ages, self.times)
```

Comment:
Is view really the right term here? This isn't providing a view into another datastructure, it is the datastructure.

Reply:
You're right. It was a view into a single dataframe, initially. Now it's not. Changed to `_PriorGrid`.